ui/lib/sync/addon/components/secrets/page/overview.ts

```diff
@@ -68,8 +67,10 @@ export default class SyncSecretsDestinationsPageComponent extends Component<Args
   });
 
   @action
-  onDocsConfirmChange(event: HTMLElementEvent<HTMLInputElement>) {
-    this.confirmDisabled = !event.target.checked;
+  closeActivateSecretsSyncModal() {
+    // need to close the modal and reset the docsConfirmChange boolean
+    this.showActivateSecretsSyncModal = false;
+    this.docsConfirmChange = false;
```

Comment:
I wonder if we should also reset `this.docsConfirmChange` in the `finally()` below 🤔 I could go both ways...If the network request errors, since we close the modal I think it makes sense to reset the checkbox.

But I also see a case for preserving the state...though I think I lean toward the former, just because we close the modal on a failure.

If you decide to go that direction then perhaps we could rename this method to `resetModal()` and reuse it in the template's close method and below in the finally. We also wouldn't need the extra comment, because it'd be pretty clear what's happening

Reply:
Great thoughts. I went with the former and renamed to `resetOptInModal`